uiautomationutilities/src/main/java/com/microsoft/identity/client/ui/automation/broker/BrokerCompanyPortal.java

```diff
@@ -88,6 +88,17 @@ public void performDeviceRegistration(@NonNull final String username,
         );
     }
 
+    public void performDeviceRegistrationForFederatedUser(@NonNull final String username,
+                                          @NonNull final String password, final boolean isFederatedUser) {
+        Logger.i(TAG, "Perform Device Registration for the given federate account..");
+        TestContext.getTestContext().getTestDevice().getSettings().addWorkAccount(
+                this,
+                username,
+                password,
+                true
```

Comment:
Why hard coding `true` here?

Reply:
changed it to use the param isFederatedUser sent in the method